src/artery/application/PersonMiddleware.h

```diff
@@ -8,18 +8,27 @@
 #define ARTERY_PERSONMIDDLEWARE_H_IH5GPCMP
 
 #include "artery/application/Middleware.h"
-#include "artery/traci/PersonMobility.h"
+#include "artery/application/VehicleDataProvider.h"
+#include "artery/traci/PersonController.h"
 
 namespace artery
 {
 
 class PersonMiddleware : public Middleware
 {
     public:
+        PersonMiddleware();
         void initialize(int stage) override;
+        void finish() override;
+
+    protected:
+        void initializeStationType();
+        void initializePersonController(omnetpp::cPar&);
+        void receiveSignal(omnetpp::cComponent*, omnetpp::simsignal_t, omnetpp::cObject*, omnetpp::cObject*) override;
 
     private:
-        PersonMobility* mMobility = nullptr;
+        traci::PersonController* mPersonController = nullptr;
+        VehicleDataProvider mVehicleDataProvider;
```

Comment:
So this is the end of `VehicleDataProvider`, isn't it? Maybe it shall become `MovingDataProvider`?

Reply:
Maybe. Persons have little to do with vehicle data, but we kept the class name because the VDP is also referenced in the CAM standard and we hesitated to remove it. However, the current `VehicleDataProvider` is providing functionalities that might be part of the Position and Time Management Entity in the Facilities Layer (ETSI EN 302 890-2). Although it does not implement all functionalities of the PoTi, it could make sense to go in that direction when renaming it.

Reply:
I see. For the moment, having a `VehicleDataProvider` in a `PersonMiddleware` is odd but not a show stopper.

Instead of aggregating several features in `VehicleDataProvider` or a PoTi entity, it might make sense to register multiple "single purpose" entities in our Facilities, e.g. a `CurvatureCalculator` (possibly different implementations depending on vehicle type), `PositionProvider` (generic), and specific entities such as `VehicleController` and `PersonController` and their respective mobility objects.

A CA VDP might then become mere syntactic sugar holding references to these single-purpose entities fetched during service initialization. I still need to wrap my mind around this… I am happy to hear your thoughts on this topic too!

Reply:
I like the idea of having a single entity that can be accessed from applications and services and provides access to the vehicle's current state, so basically the VDP. For me, it is intuitive that way.

IMO, having many fractioned entities in the facilities makes it more difficult to use and requires lots of changes in the current architecture.

Maybe a generalized DataProvider with more specific classes for vehicles, persons, bicycles, and such, could be a way forward.  Another option could be to have a generic DataProvider class and move specific functions, such as curvature, into the Kinematics class for the respective vehicle type.